### PR TITLE
feat(replace NATS): add nexus and child ops over grpc

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
- Change transport for nexus and child ops from rest over gRPC replacing NATS.
- Fix corresponding tests

Signed-off-by: Abhinandan Purkait <abhinandan.purkait@mayadata.io>

### CODE REVIEW

**Comment by Abhinandan-Purkait** [comment]
bors try

**Comment by Abhinandan-Purkait** [comment]
bors cancel

**Comment by Abhinandan-Purkait** [comment]
bors try

**Comment by Abhinandan-Purkait** [comment]
bors try

**Comment by Abhinandan-Purkait** [comment]
bors try-

**Comment by Abhinandan-Purkait** [comment]
bors try

File: control-plane/grpc/src/operations/nexus/traits.rs [hunk 1]
```diff
@@ -13,6 +24,49 @@ use common_lib::{
 };
 use std::convert::TryFrom;
 
+/// All nexus operations to be a part of the ReplicaOperations trait
```
Comment (tiagolobocastro) [inline control-plane/grpc/src/operations/nexus/traits.rs:27]: ReplicaOperations - NexusOperations ?

**Comment by Abhinandan-Purkait** [comment]
bors try

File: common/src/types/v0/message_bus/nexus.rs [hunk 2]
```diff
@@ -200,6 +200,10 @@ impl NvmfControllerIdRange {
         const MAX_CONTROLLER_ID: u16 = 0xffef;
         MIN_CONTROLLER_ID ..= MAX_CONTROLLER_ID
     }
+    /// create a new NvmfControllerIdRange using provided range
+    pub fn new(start: u16, end: u16) -> Self {
```
Comment (tiagolobocastro) [inline common/src/types/v0/message_bus/nexus.rs:204]: This should return an error, we should validate if the range is within 1..=0xffef ?

**Review by tiagolobocastro — APPROVED** [review]

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**